### PR TITLE
Apply PHP Version to use in composer.json in rector.php for php version features

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

File: packages/NodeTypeResolver/ValueObject/OldToNewType.php [hunk 1]
```diff
@@ -9,18 +9,18 @@
 final class OldToNewType
 {
     public function __construct(
-        private Type $oldType,
-        private Type $newType
+        private Type $type,
+        private Type $type
```
Comment (samsonasik) [inline packages/NodeTypeResolver/ValueObject/OldToNewType.php]: found a bug for multiple same type got same variable

File: packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php [hunk 2]
```diff
@@ -46,8 +50,9 @@ public function resolveAnnotationMethodCall(BetterTokenIterator $tokenIterator):
      * @see https://github.com/doctrine/annotations/blob/c66f06b7c83e9a2a7523351a9d5a4b55f885e574/lib/Doctrine/Common/Annotations/DocParser.php#L1215-L1224
      * @return mixed|mixed[]
      */
-    public function resolveAnnotationValue(BetterTokenIterator $tokenIterator)
-    {
+    public function resolveAnnotationValue(
+        BetterTokenIterator $tokenIterator
+    ): CurlyListNode | string | array | array | ConstExprFalseNode | ConstExprTrueNode | ConstExprIntegerNode | DoctrineAnnotationTagValueNode | array {
```
Comment (samsonasik) [inline packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php]: found a bug for multiple piped array type

File: packages/NodeTypeResolver/ValueObject/OldToNewType.php [hunk 3]
```diff
@@ -16,11 +16,11 @@ public function __construct(
...
-        return $this->oldType;
+        return $this->type;
     }
 
     public function getNewType(): Type
     {
-        return $this->newType;
+        return $this->type;
     }
 }
```
Comment (TomasVotruba) [inline packages/NodeTypeResolver/ValueObject/OldToNewType.php]: This is a bug

File: rector.php [hunk 4]
```diff
@@ -112,6 +112,6 @@
     ]);
 
     $parameters->set(Option::PHPSTAN_FOR_RECTOR_PATH, __DIR__ . '/phpstan-for-rector.neon');
-    $parameters->set(Option::PHP_VERSION_FEATURES, PhpVersion::PHP_74);
+    $parameters->set(Option::PHP_VERSION_FEATURES, PhpVersion::PHP_80);
     $parameters->set(Option::ENABLE_CACHE, true);
```
Comment (TomasVotruba) [inline rector.php:115]: :+1: I think we can remove this as `composer.json` php versoin should be used by default

Comment (samsonasik) [reply rector.php:115]: remove `PHP_VERSION_FEATURES` constant?

File: packages/NodeTypeResolver/ValueObject/OldToNewType.php (hunk 3 above)
Comment (samsonasik) [reply packages/NodeTypeResolver/ValueObject/OldToNewType.php]: yes, need to be checked one of these example:

```diff
rules/CodingStyle/ValueObject/NameAndParent.php
    ---------- begin diff ----------
--- Original
+++ New
@@ @@
      * @param Name|Identifier $nameNode
      */
     public function __construct(
-        private Node $nameNode,
-        private Node $parentNode
+        private Node $node,
+        private Node $node
     ) {
     }

-    /**
-     * @return Name|Identifier
-     */
-    public function getNameNode(): Node
+    public function getNameNode(): Name|Identifier
     {
-        return $this->nameNode;
+        return $this->node;
     }

     public function getParentNode(): Node
     {
-        return $this->parentNode;
+        return $this->node;
     }
 }
    ----------- end diff -----------

Applied rules:
 * ArgumentAdderRector
 * UnionTypesRector
 * ReturnTypeDeclarationRector (https://wiki.php.net/rfc/scalar_type_hints_v5)
 ```

Comment (samsonasik) [reply packages/NodeTypeResolver/ValueObject/OldToNewType.php]: It seems on `RenamePropertyToMatchTypeRector`

File: rector.php (hunk 4 above)
Comment (samsonasik) [reply rector.php:115]: I removed the `$parameters->set(Option::PHP_VERSION_FEATURES, PhpVersion::PHP_80);` https://github.com/rectorphp/rector-src/pull/159/commits/16cc001039be4f36c52bdae789e65f2d73624ac7

**Comment by TomasVotruba** [comment]
It's too many changes at once. It would be better to split it per rule PRs.

**Comment by samsonasik** [comment]
Ok, the error is only in `PHPStan`, which means the rules seems working ok.

**Pull request closed without merging**